src/components/PostDetails.tsx

```diff
@@ -1,60 +1,37 @@
 import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { AnyAction, ThunkDispatch } from '@reduxjs/toolkit';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
 import * as commentsApi from '../api/comments';
 
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import { useAppSelector } from '../app/hooks';
+import { commentsSlice, fetchComments } from './Comments';
+import { RootState } from '../app/store';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
   const [visible, setVisible] = useState(false);
 
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
+  const comments2 = useAppSelector(state => state.comments.value);
+  const hasError2 = useAppSelector(state => state.comments.error);
+  const loaded2 = useAppSelector(state => !state.comments.isLoading);
 
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
-    setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
-    }
-  };
+  const dispatch = useDispatch();
+  const thunkDispatch: ThunkDispatch<
+  RootState, unknown, AnyAction> = useDispatch();
 
   useEffect(() => {
-    loadComments();
-  }, []);
+    setVisible(false);
 
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+    thunkDispatch(fetchComments(post.id));
+  }, [post]);
```

Comment:
<img width="910" alt="image" src="https://github.com/mate-academy/react_redux-list-of-posts/assets/64750958/4c190d3d-2644-460b-b427-b8eabe71f4af">